services/platform/platform.go

```diff
@@ -112,34 +117,119 @@ func (s *Service) SignUp(ctx context.Context, email, firstName, lastName string)
 	return nil
 }
 
-// SignIn checks Percona Platform user authentication and creates session.
-func (s *Service) SignIn(ctx context.Context, email, password string) error {
-	cc, err := saasdial.Dial(ctx, "", s.host)
+// Connect checks if PMM is connected. If it's not, it connects the a PMM server to the Portal.
+func (s *Service) Connect(ctx context.Context, serverName, email, password string) error {
+	_, err := models.GetPerconaSSODetails(s.db.Querier)
+	if err == nil {
+		return errors.Wrap(err, "PMM server is already connected to Portal")
+	}
+	settings, err := models.GetSettings(s.db.Querier)
 	if err != nil {
-		return errors.Wrap(err, "failed establish connection with Percona")
+		return errors.Wrap(err, "failed to fetch PMM server ID and address of PMM server")
```

Comment:
nit: `failed to fetch PMM server ID and address` should suffice

Reply:
done